Fish update just dropped. New fish, aquarium type, two traits and a research node.

tralezab opened the pull request:
## About The Pull Request

### New fish

#### anxious zipzap
An incredibly anxiety-ridden and electroreceptive fish. Worried about the walls of its tank closing in constantly. Both literally and as a general metaphorical unease about life's direction.

It produces electricity, and HATES being with any other fish. One other zipzap is allowed at most.

![image](https://github.com/tgstation/tgstation/assets/40974010/6a1a31f3-1c84-4b5e-b8ca-652c5007813a)

#### monocloning jumpercable
A surprisingly useful if nasty looking creation from the syndicate fish labs. Drop one in a tank, and watch it self-feed and multiply. Generates more and more power as a growing swarm!

A far better option for power generation than the zipzap, but it's a syndicate fish. Get an emag or get very lucky!

![image](https://github.com/tgstation/tgstation/assets/40974010/7f4aaf32-919c-40d9-890f-72ea9f91ac3c)

Working spacebase example

![image](https://github.com/tgstation/tgstation/assets/40974010/8eeaaf18-c0ab-48a5-8ada-e0eb7f118dd9)


### Aquarium type

Bioelectricity Generator! It boosts the power of electroreceptive fish to very dangerous levels! But, uh, you put a tesla coil, and voila! Free power! ...Free power if you keep the fish fed! More fish, more power!

![image](https://github.com/tgstation/tgstation/assets/40974010/e6abcd5b-93df-4ffc-ac6a-78585fd4a6c6)

You can convert an aquarium into a bioelectricity generator by researching the new node, "Marine Utility", from bio processing. Then you build the biogen upgrade kit and apply it to an already existing aquarium.

### Fish Traits

#### Electrogenesis

Trait turns the fish into a shock weapon while it's alive, and every time it feeds, it releases a weak tesla bolt to anyone close to the tank.

#### Anxiety

anxious fish die when any other fish are in the tank. Please keep them alone :[

## Why It's Good For The Game

Fish are great, but fish don't have anything to give back to the station, right? Well, there's some small stuff, but I wanted to add a bit of functionality towards being able to benefit from having fish.

The anxiety trait with the zipzap gives spessmen a bit of a challenge. You can settle for one fish generating power per tank and do a multi-tank generation setup, or you can do some crossbreeding to make a fish WITH electroreceptive but WITHOUT anxiety. Neat!

## Changelog
:cl:
add: New fish, the anxious zipzap
add: New syndicate fish, the monocloning jumpercable
add: New aquarium, the bioelectricity generator
fix: Mixotrophic fish now properly lack food requirements
/:cl:

tralezab committed:
ok

tralezab committed:
jumpercable added

tralezab committed:
my bad, you are a zipzap i should know that

tralezab committed:
linter, missing sprites, fixes

Higgin commented:
Why is it that when virology is on the ropes, fish begin to rise? It's like the fourth time in the last three years.

this is awesome btw, and great quality on the sprites

GuillaumePrata commented:
Now we remove the SM from Icebox as it has unfixable design issues on that enviroment and replace it with a big frostelectric fish.

Time to save engineering!

Jacquerel commented on code/modules/fishing/aquarium/aquarium.dm:
```diff
@@ -212,6 +216,20 @@
+			moving.forceMove(upgraded_aquarium)
+		balloon_alert(user, "upgraded")
+		qdel(upgrade)
+		qdel(src)
```
presumably we want to return here

Jacquerel commented on code/modules/fishing/aquarium/aquarium_upgrades.dm:
```diff
@@ -0,0 +1,37 @@
+	var/explosive = zap_flags & ZAP_MACHINE_EXPLOSIVE
+	if(!explosive)
+		return //immune to all other shocks to make sure power can be generated without breaking the generator itself
+	. = ..()
```
```suggestion
	return ..()
```

necromanceranne commented on code/modules/fishing/fish/fish_traits.dm:
```diff
@@ -379,3 +379,60 @@ GLOBAL_LIST_INIT(fish_traits, init_subtypes_w_path_keys(/datum/fish_trait, list(
+/datum/fish_trait/electrogenesis/proc/on_fish_fed(obj/item/fish/fish, seconds_per_tick)
+	SIGNAL_HANDLER
+	var/fish_zap_range = 1
+	var/fish_zap_power = 5000
```
Are there not defines for zap strength?

tralezab replied:
I'd like to use them if there is, but everywhere i looked there wasn't. I just ran the calculations myself. The non-bioelec tank shock is about 8 damage, the bioelec tank shock is 33. Would be nice if we had a define that converted a damage amount to the number required to cause it

tralezab replied:
There was joules which kinda helps

tralezab committed:
fixes electrogenesis, flavor

tralezab committed:
testing and fixes from testing

tralezab committed:
small fluff update

tralezab committed:
refined power balance, reworked anxiety

anxiety now only allows 3 fish total, of any kind. this will let people actually crossbreed the fish. zipzap generates double power, as you cannot stack it very well in one tank.

CCC23 commented:
time to make fish chernobyle as a tot with 30000 jumper cables  30 fish tanks and enough electricity production to vaporize anyone even near a power cable

san7890 commented on code/modules/fishing/fish/fish_types.dm:
```diff
@@ -612,3 +632,26 @@
+		/datum/fish_trait/no_mating,
+		/datum/fish_trait/wary,
+		/datum/fish_trait/anxiety,
+		/datum/fish_trait/electrogenesis
```
```suggestion
		/datum/fish_trait/electrogenesis,
```

san7890 commented on code/modules/research/designs/power_designs.dm:
```diff
@@ -101,6 +101,16 @@
+	id = "bioelec_gen"
+	build_path = /obj/item/aquarium_upgrade/bioelec_gen
+	category = list(
+		RND_CATEGORY_MACHINE + RND_SUBCATEGORY_MACHINE_ENGINEERING
```
```suggestion
		RND_CATEGORY_MACHINE + RND_SUBCATEGORY_MACHINE_ENGINEERING,
```

san7890 commented on code/modules/fishing/fish/fish_traits.dm:
```diff
@@ -379,3 +379,53 @@ GLOBAL_LIST_INIT(fish_traits, init_subtypes_w_path_keys(/datum/fish_trait, list(
+	if(!fish.loc || fish.status == FISH_DEAD)
+		return
+	for(var/obj/item/other_fish in fish.loc.contents)
+		if(!fish_tolerance)
```
```suggestion
		if(fish_tolerance == 0)
```

kindly do not treat integers as booleans

san7890 commented on code/modules/fishing/fish/fish_types.dm:
```diff
@@ -295,6 +295,26 @@
+	fish_traits = list(
+		/datum/fish_trait/parthenogenesis,
+		/datum/fish_trait/mixotroph,
+		/datum/fish_trait/electrogenesis
```
```suggestion
		/datum/fish_trait/electrogenesis,
```

san7890 reviewed (commented):
bless my beautiful trailing commas in lists life (and other nit)

Admiral-Gigaton commented:
> time to make fish chernobyle as a tot with 30000 jumper cables 30 fish tanks and enough electricity production to vaporize anyone even near a power cable

not a bad idea... how much fish fits in an aquarium, or one of these electric aquariums?

tralezab commented:
oh its totally possible with the syndicate fish, if you can get your hands on it. 20 fish maximum in an aquarium, cables multiply up to 12 so you'll need a separate breeding tank to get past 12 soft cap, then you consider multiple tanks and yeaahhhh you can make something hilarious

tralezab committed:
review

tralezab commented:
review handled, ready 👍
